src/GitVersionCore/Configuration/ConfigurationProvider.cs

```diff
@@ -144,14 +144,28 @@ public static string GetEffectiveConfigAsString(string gitDirectory, IFileSystem
             return stringBuilder.ToString();
         }
 
-        static string GetConfigFilePath(string workingDirectory)
+        static string GetConfigFilePath(string workingDirectory, IFileSystem fileSystem)
         {
-            return Path.Combine(workingDirectory, "GitVersionConfig.yaml");
+            var ymlPath = Path.Combine(workingDirectory, "GitVersion.yml");
+            if (fileSystem.Exists(ymlPath))
+            {
+                return ymlPath;
+            }
+
+            var yamlPath = Path.Combine(workingDirectory, "GitVersionConfig.yaml");
+            if (fileSystem.Exists(yamlPath))
+            {
+                Logger.WriteWarning("'GitVersionConfig.yaml' is deprecated, use 'GitVersion.yml' instead.");
+
+                return yamlPath;
+            }
+
+            return ymlPath;
```

Comment:
Hi Pascal,
The last two returns are not the same, the complete logic is:

- if the new file GitVersion.yml already exists we return it
- if the deprecated file GitVersionConfig.yaml already exists we warn and return it
- if no file exists we return the new file GitVersion.yml